File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.java
class Solution {
    public int solution(int[] array) {
        int answer = 0;
        int[] count = new int[1001];
        int max = Integer.MIN_VALUE;

        for(int i = 0; i < array.length; i++) count[array[i]]++;
        
        for(int j = 0; j < count.length; j++) {
            if(count[j] > max) {
                max = count[j];
                answer = j;
            }else if(count[j] == max){
                answer = -1;
            }
        }

        return answer;
    }
}

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// array_len은 배열 array의 길이입니다.
int solution(int array[], size_t array_len) {
    int answer = 0;
    int maxCount = 0;
    int maxValue = 0;
    int countArray[1000] = {0,};

    for(int i = 0; i < array_len; i++){
        countArray[array[i]]++;
        if(maxValue < array[i]){
            maxValue = array[i];
        }
    }
    
    for(int j = 0; j <= maxValue; j++){
        if(countArray[j] == 0) continue;
        
        if(maxCount < countArray[j]){
            maxCount = countArray[j];
            answer = j;
        }else if(maxCount == countArray[j]){
            answer = -1;
        }
    }
    
    return answer;
}